parser/prism/Translator.cc

```diff
@@ -0,0 +1,2011 @@
+            auto aliasMethodNode = down_cast<pm_alias_method_node>(node);
+
+            auto newName = translate(aliasMethodNode->new_name);
+            auto oldName = translate(aliasMethodNode->old_name);
+
+            return make_unique<parser::Alias>(location, move(newName), move(oldName));
+        }
+        case PM_AND_NODE: { // operator `&&` and `and`
+            auto andNode = down_cast<pm_and_node>(node);
+
+            auto left = translate(andNode->left);
+            auto right = translate(andNode->right);
+
+            return make_unique<parser::And>(location, move(left), move(right));
+        }
+        case PM_ARGUMENTS_NODE: { // A list of arguments in one of several places:
+            // 1. The arguments to a method call, e.g the `1, 2, 3` in `f(1, 2, 3)`.
+            // 2. The value(s) returned from a return statement, e.g. the `1, 2, 3` in `return 1, 2, 3`.
+            // 3. The arguments to a `yield` call, e.g. the `1, 2, 3` in `yield 1, 2, 3`.
+            unreachable("PM_ARGUMENTS_NODE is handled separately in `Translator::translateArguments()`.");
+        }
+        case PM_ARRAY_NODE: { // An array literal, e.g. `[1, 2, 3]`
+            auto arrayNode = down_cast<pm_array_node>(node);
+
+            auto sorbetElements = translateMulti(arrayNode->elements);
+
+            return make_unique<parser::Array>(location, move(sorbetElements));
+        }
+        case PM_ASSOC_NODE: { // A key-value pair in a Hash literal, e.g. the `a: 1` in `{ a: 1 }
+            auto assocNode = down_cast<pm_assoc_node>(node);
+
+            auto key = translate(assocNode->key);
+            auto value = translate(assocNode->value);
+
+            return make_unique<parser::Pair>(location, move(key), move(value));
+        }
+        case PM_ASSOC_SPLAT_NODE: { // A Hash splat, e.g. `**h` in `f(a: 1, **h)` and `{ k: v, **h }`
+            unreachable("PM_ASSOC_SPLAT_NODE is handled separately in `Translator::translateKeyValuePairs()` and "
+                        "`PM_HASH_PATTERN_NODE`, because its translation depends on whether its used in a "
+                        "Hash literal, Hash pattern, or method call.");
+        }
+        case PM_BACK_REFERENCE_READ_NODE: {
+            auto backReferenceReadNode = down_cast<pm_back_reference_read_node>(node);
+            auto name = parser.resolveConstant(backReferenceReadNode->name);
+
+            return make_unique<parser::Backref>(location, gs.enterNameUTF8(name));
+        }
+        case PM_BEGIN_NODE: { // A `begin ... end` block
+            auto beginNode = down_cast<pm_begin_node>(node);
+
+            NodeVec statements;
+            unique_ptr<parser::Node> translatedRescue;
+
+            if (beginNode->rescue_clause != nullptr) {
+                // Extract rescue and else nodes from the begin node
+                auto bodyNode = translateStatements(beginNode->statements, true);
```

Comment:
I really like the `auto inlineIfSingle = true;` variables and their uses in other places to give the reader a sense of what the `true` parameter to `translateStatements` means.  Could that be done here as well?

And in a number of other places throughout this file?

(I'm honestly not so sure that `inlineIfSingle` shouldn't be an optional parameter on `translateStatements` that defaults to `true`, and then the one `false` would be a little more obvious.)